conn.go

```diff
@@ -942,16 +1048,22 @@ func (c *Conn) readLoop(name string, pktConn ipPacketConn, inboundBufferSize int
 				for _, query := range queries {
 					queryCopy := query
 					if queryCopy.nameWithSuffix == a.Name.String() {
-						ip, err := ipFromAnswerHeader(a, p)
+						addr, err := addrFromAnswerHeader(a, p)
 						if err != nil {
-							c.log.Warnf("failed to parse mDNS answer %v", err)
+							c.log.Warnf("[%s] failed to parse mDNS answer %v", c.name, err)
 							return
 						}
 
+						resultAddr := *addr
+						// DNS records don't contain IPv6 zones.
```

Comment:
Calling this out as being important to getting the properly attributed zone back